tests/structures/logic/test_z3_logic_converter.py

```diff
@@ -0,0 +1,55 @@
+import pytest as pytest
+from decompiler.structures.pseudo import BinaryOperation, Branch, Condition, Constant, OperationType, Variable
+from decompiler.structures.pseudo.logic import BaseConverter
+from decompiler.structures.pseudo.z3_logic import Z3Converter
+from z3 import BoolRef
+
+
+def _generate_instr_bool_as_numbers(op: OperationType) -> Branch:
+    return Branch(
+        Condition(
+            OperationType.not_equal,
+            [
+                BinaryOperation(
+                    op,
+                    [
+                        Condition(OperationType.greater_us, [Variable("a"), Constant(64)]),
+                        Condition(OperationType.less_or_equal_us, [Variable("b"), Constant(63)]),
```

Comment:
I think this can never be a condition and should be a `BinaryOperation`. Should not change much, but I think we never construct this, at leas I hope so.

Reply:
Yeah I hope so too, but was not sure which cases I would need to cover to get all invalid `BOOLREF` ops. 
I took every operation which i though was somehow likely (even tho it is very unlikely, because the error only occurred for one binary, because somehow a cast was missing; but HEY). 

Should I left them all in place? Or remove a few conditions from `OPERATIONS_INVALID_BOOLREF_OP`?
I mean both options are fine: 
If I remove a few and the error will arise again we can fix it very quickly, or if not, we know (/or no know not), that the code fixes the problem.

Reply:
I think i will remove the logic ones and only keep the numerical ones. 
If for some reason another issue like this arises I'll re-add them.